sql/core/src/test/scala/org/apache/spark/sql/DataFrameSetOperationsSuite.scala

```diff
@@ -387,100 +388,104 @@ class DataFrameSetOperationsSuite extends QueryTest with SharedSparkSession {
 
   test("SPARK-34283: SQL-style union using Dataset, " +
     "remove unnecessary deduplicate in multiple unions") {
-    val unionDF = testData.union(testData).distinct().union(testData).distinct()
-      .union(testData).distinct().union(testData).distinct()
-
-    // Before optimizer, there are three 'union.deduplicate' operations should be combined.
-    assert(unionDF.queryExecution.analyzed.collect {
-      case u: Union if u.children.size == 4 => u
-    }.size === 1)
-
-    // After optimizer, four 'union.deduplicate' operations should be combined.
-    assert(unionDF.queryExecution.optimizedPlan.collect {
-      case u: Union if u.children.size == 5 => u
-    }.size === 1)
-
-    checkAnswer(
-      unionDF.agg(avg("key"), max("key"), min("key"),
-        sum("key")), Row(50.5, 100, 1, 5050) :: Nil
-    )
+    withSQLConf(SQLConf.OPTIMIZER_EXCLUDED_RULES.key -> RemoveNoopOperators.ruleName) {
```

Comment:
This seems to reduce the test coverage for the existing other rules of `RemoveNoopOperators`. Can we make a new rule for this PR instead of touching `RemoveNoopOperators`?

Reply:
Sure.